packages/pds/src/sql-blockstore.ts

```diff
@@ -0,0 +1,49 @@
+import { IpldStore } from '@atproto/repo'
+import { CID } from 'multiformats/cid'
+import Database from './db'
+
+export class SqlBlockstore extends IpldStore {
+  constructor(public db: Database, public did: string) {
+    super()
+  }
+
+  async has(cid: CID): Promise<boolean> {
+    const found = await this.db.db
+      .selectFrom('ipld_block')
+      .where('cid', '=', cid.toString())
+      .select('cid')
+      .executeTakeFirst()
+    return !!found
+  }
+
+  async getBytes(cid: CID): Promise<Uint8Array> {
+    const found = await this.db.db
+      .selectFrom('ipld_block')
+      .where('cid', '=', cid.toString())
+      .select('content')
+      .executeTakeFirst()
+    if (!found) throw new Error(`Not found: ${cid.toString()}`)
+    return found.content
+  }
+
+  async putBytes(cid: CID, bytes: Uint8Array): Promise<void> {
+    const row = {
+      cid: cid.toString(),
+      did: this.did,
+      size: bytes.length,
+      content: bytes,
+      indexedAt: new Date().toISOString(),
```

Comment:
Random thought here: it would be kind of nice if rows written in the same transaction had the same `indexedAt` times.  If we want to achieve that with the blockstore, the the constructor for `SqlBlockstore` could received a date, and if we're in a transaction then that date would be used for `indexedAt`.  In all other cases, it would work as it currently does.

Reply:
May also be a little funky.  Honestly no strong feelings on it, especially if we're nixing the `indexedAt` column on the records table anyway.

Reply:
yeah that's not a bad idea. could apply that idea to other transactional operations as well 🤔